crates/noirc_frontend/src/graph/mod.rs

```diff
@@ -54,11 +60,28 @@ impl FromStr for CrateName {
     type Err = String;
 
     fn from_str(name: &str) -> Result<Self, Self::Err> {
-        let is_invalid = name.chars().any(|n| CHARACTER_BLACK_LIST.contains(&n));
-        if is_invalid {
-            Err(name.into())
-        } else {
+        if Self::is_valid_name(name) {
             Ok(Self(SmolStr::new(name)))
+        } else {
+            Err("Package names must be non-empty and cannot contain hyphens".into())
```

Comment:
I like bringing in this better error message, but should we also inject the package name? Something like:
```suggestion
            Err(format!("Package names must be non-empty and cannot contain hyphens, but received {name}").into())
```

Reply:
Clap displays the bad package name by default.

Reply:
When we resolve a workspace, we map the string error to a variant and it doesn't contain the name at all. I'm going to push some updates that make this stuff more clear to this branch